clang/lib/Interpreter/Interpreter.cpp

```diff
@@ -760,8 +787,10 @@ std::unique_ptr<llvm::Module> Interpreter::GenModule() {
   return nullptr;
 }
 
-CodeGenerator *Interpreter::getCodeGen() const {
-  FrontendAction *WrappedAct = Act->getWrapped();
+CodeGenerator *Interpreter::getCodeGen(IncrementalAction *Action) const {
+  if (!Action)
+    Action = Act.get();
+  FrontendAction *WrappedAct = Action->getWrapped();
```

Comment:
I do not understand this change. Is it because we call `getCodeGen` from both Device and Host compiler instance but we only return the codegen for the Host?

Reply:
Yupp. Basically this comment https://github.com/llvm/llvm-project/pull/136404#discussion_r2058443908

Reply:
Shouldn't we delegate somehow to the IncrementalParser/IncrementalCUDAParser instead?